packages/plugin-typescript/sources/typescriptUtils.ts

```diff
@@ -0,0 +1,58 @@
+import {Request, Requester, Response} from '@algolia/requester-common';
+import {Configuration, Descriptor}    from '@yarnpkg/core';
+import {httpUtils, structUtils}       from '@yarnpkg/core';
+import algoliasearch                  from 'algoliasearch';
+
+// Note that the appId and appKey are specific to Yarn's plugin-typescript - please
+// don't use them anywhere else without asking Algolia's permission
+const ALGOLIA_API_KEY = 'e8e1bd300d860104bb8c58453ffa1eb4';
+const ALGOLIA_APP_ID = 'OFCNCOG2CU';
+
+interface AlgoliaObj {
+  types?: {
+    ts?: string
+  }
+}
+
+export const hasDefinitelyTyped = async (
+  descriptor: Descriptor,
+  configuration: Configuration,
+) => {
+  const stringifiedIdent = structUtils.stringifyIdent(descriptor);
+  const algoliaClient = createAlgoliaClient(configuration);
+  const index = algoliaClient.initIndex('npm-search');
+
+  try {
+    const packageInfo = await index.getObject<AlgoliaObj>(stringifiedIdent, {attributesToRetrieve: ['types']});
+
+    return packageInfo.types?.ts === 'definitely-typed';
+  } catch (_e) {
+    return false;
+  }
+};
+
+const createAlgoliaClient = (configuration: Configuration) => {
```

Comment:
It might make sense to add this to its own file, in case we want to e.g. provide CLI searching?

Reply:
For now let's keep it there; the `typescript` plugin shouldn't expose the Algolia logic, and apart from moving that into the core we don't have a good place to put this. It's ok to have some duplication for now and see where that leads us 🙂